bennu-oauth/src/main/java/org/fenixedu/bennu/oauth/domain/ExternalApplication.java

```
@@ -85,10 +88,37 @@ public boolean matchesUrl(String redirectUrl) {
     public boolean matchesSecret(String secret) {
         return !Strings.isNullOrEmpty(secret) && secret.equals(getSecret());
     }
+    
+    public boolean matchesCodeVerifier(String codeVerifier, String code) {
+    	MessageDigest digester = null;
+    	try {
+    		digester = MessageDigest.getInstance("SHA-256");
+
+    	} catch (Exception e) {
+    		e.printStackTrace();
```

Should probably throw an Error(e) in this case...